Jenkinsfile

```
@@ -61,11 +61,32 @@ node('linux') {
 
     if (infra.isTrusted()) {
         stage('Publish') {
-            sh 'mkdir -p updates'
-            sh 'cp target/*.json target/*.html updates'
+            sh '''
+                mkdir -p updates
+                cp target/*.json target/*.html updates
+            '''
             sshagent(['updates-rsync-key']) {
                 sh 'rsync -avz  -e \'ssh -o StrictHostKeyChecking=no\' --exclude=.svn updates/ www-data@updates.jenkins.io:/var/www/updates.jenkins.io/updates/'
             }
+            withCredentials([
+                string(credentialsId: 'updates-jenkins-io-file-share-url-with-sas-token', variable: 'UPDATES_FILE_SHARE_URL'),
+                string(credentialsId: 'aws-access-key-id-updatesjenkinsio', variable: 'AWS_ACCESS_KEY_ID'),
+                string(credentialsId: 'aws-secret-access-key-updatesjenkinsio', variable: 'AWS_SECRET_ACCESS_KEY')
+            ]) {
+                sh '''
+                ls -al updates/
+
+                azcopy sync updates/ "${UPDATES_FILE_SHARE_URL}" --exclude-path '.svn' --recursive=true --delete-destination=true
+
+                ## Note: AWS CLI are configured through environment variables (from Jenkins credentials) - https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
+                aws s3 sync updates/ s3://"${UPDATES_R2_BUCKETS}"/ \
```

You need to specify the AWS region as the (ephemeral) agent won't have any `$HOME/.aws` configured setup, otherwise the command will fail.

Example in https://github.com/jenkins-infra/update-center2/blob/be3e96b613d1fe6743cf569d21577e425baf312c/site/publish.sh#L11.

```suggestion
                export AWS_DEFAULT_REGION=auto
                aws s3 sync updates/ s3://"${UPDATES_R2_BUCKETS}"/ \
```